lib/elixir/lib/protocol.ex

```diff
@@ -681,6 +683,10 @@ defmodule Protocol do
   def __protocol__(name, do: block) do
     quote do
       defmodule unquote(name) do
+        @before_compile Protocol.Util
```

Comment:
Please define the callback in the same module, we don't need a new one. :)

Reply:
Doesn't beore_compile callback impls need to be outside a module since it will not be available before compile such module is compiled?

Reply:
Nevermind. I see what you mean. This is inside a macro that outputs "defmodule", I will move it inside